src/docfx/build/page/BuildPage.cs

```diff
@@ -220,75 +240,58 @@ private static (List<Error> errors, JObject model, InputMetadata inputMetadata)
             var (schemaTransformError, transformedToken) = schemaTemplate.JsonSchemaTransformer.TransformContent(file, context, obj);
             errors.AddRange(schemaTransformError);
 
-            var (metaErrors, inputMetadata) = context.MetadataProvider.GetMetadata(file);
-            errors.AddRange(metaErrors);
-
             var pageModel = (JObject)transformedToken;
             if (file.Docset.Legacy && TemplateEngine.IsLandingData(file.Mime))
             {
-                // TODO: remove schema validation in ToObject
-                var (_, content) = JsonUtility.ToObject(transformedToken, typeof(LandingData));
-
-                // merge extension data to metadata in legacy model
-                var landingData = (LandingData)content;
-                JsonUtility.Merge(inputMetadata.RawJObject, landingData.ExtensionData);
+                var landingData = pageModel.ToObject<LandingData>();
 
                 pageModel = JsonUtility.ToJObject(new ConceptualModel
                 {
-                    Conceptual = HtmlUtility.LoadHtml(await RazorTemplate.Render(file.Mime, content)).HtmlPostProcess(file.Docset.Culture),
-                    Title = inputMetadata.Title ?? obj?.Value<string>("title"),
+                    Conceptual = HtmlUtility.LoadHtml(await RazorTemplate.Render(file.Mime, landingData)).HtmlPostProcess(file.Docset.Culture),
+                    Title = obj?.Value<string>("title"),
                     RawTitle = $"<h1>{obj?.Value<string>("title")}</h1>",
+                    ExtensionData = landingData.ExtensionData,
                 });
             }
 
-            return (errors, pageModel, inputMetadata);
+            return (errors, pageModel);
         }
 
-        private static (object model, JObject metadata) ApplyPageTemplate(Context context, Document file, JObject pageMetadata, JObject pageModel, bool isConceptual)
+        private static (TemplateModel model, JObject metadata) TransformToTemplateModel(Context context, JObject pageModel, Document file)
```

Comment:
consider `CreateXXX` to align with other `CreateXXX` methods

```suggestion
        private static (TemplateModel model, JObject metadata) CreateTemplateModel(Context context, JObject pageModel, Document file)
```